src/http_server.cc

```diff
@@ -3055,11 +3055,13 @@ HTTPAPIServer::StartTrace(
     TRITONSERVER_InferenceTrace** triton_trace)
 {
 #ifdef TRITON_ENABLE_TRACING
+  HttpTextMapCarrier carrier(req->headers_in);
```

Comment:
do these carriers just record all the headers associated with gRPC/HTTP requests?

Reply:
This one is specific to `HTTP` requests. `Carries` are initialized with headers/metadata, and then OTel propagator will use them to extract relevant headers. 
In this comment I provide a deep dive of what is going on:
https://github.com/triton-inference-server/server/pull/6785#discussion_r1449664084